test/why.js

```diff
@@ -359,7 +359,7 @@ test('functions', function (t) {
 		var genFnSpaceStar = Function('return function *() {};')();
 		var genNoSpaces = Function('return function*(){};')();
 		st.equal(
-			'second argument is a Generator; first is not',
+			'toStringTag is not the same: [object Function] !== [object GeneratorFunction]',
```

Comment:
What I'd rather do here is something like:
```js
var reasonsMap = {
  'second argument is a Generator; first is not': true,
  'toStringTag is not the same: [object Function] !== [object GeneratorFunction]': true
};
var reasons = objectEntries(reasonsMap);
var actual = isEqualWhy(fnNoSpace, genNoSpaces);
reasonsMap[actual] = true;
st.deepEqual(objectEntries(reasonsMap), reasons, 'generator and fn that are otherwise identical are not equal');
```

This seems to make tests pass for me on node 5 and 6.